### PR TITLE
feat(docs): document `tickscript` package

### PR DESCRIPTION
**Pull request by pierwill**

### CODE REVIEW

File: stdlib/contrib/bonitoo-io/tickscript/tickscript.flux [hunk 1]
```diff
@@ -118,12 +247,41 @@ select = (column="_value", fn=(column, tables=<-) => tables, as, tables=<-) => {
...
+// selectWindow changes a column’s name, windows rows by time,
+// and applies an aggregate or selector function the specified column for each window of time.
+//
+// ## TICKscript helper function
+// `tickscript.selectWindow` is a helper function meant to replicate TICKscript operations like the following:
+// ```
+// Rename, window, and aggregate
+// query("SELECT f(x) AS y")
+//   .groupBy(time(t), ...)
+// ```
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/tickscript/tickscript.flux:259]:
```suggestion
// selectWindow changes a column’s name, windows rows by time, and then applies an
// aggregate or selector function the specified column for each window of time.
//
// ## TICKscript helper function
// `tickscript.selectWindow` is a helper function meant to replicate TICKscript operations like the following:
//
// ```
// // Rename, window, and aggregate
// query("SELECT f(x) AS y")
//   .groupBy(time(t), ...)
// ```
```

File: stdlib/contrib/bonitoo-io/tickscript/tickscript.flux [hunk 2]
```diff
@@ -118,12 +247,41 @@ select = (column="_value", fn=(column, tables=<-) => tables, as, tables=<-) => {
...
+// ## Parameters
 //
-//   query("SELECT f(x) AS y")
-//     .groupBy(time(t), ...)
+// - column: - string - Column to operate on. Default is _value.
+// - fn: - function - (Required) Aggregate or selector function to apply.
+// - as: - string - (Required) New column name.
+// - every: - duration - (Required) Duration of windows.
+// - defaultValue: (Required) Default fill value for null values in column.
+//   Must be the same data type as column.
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/tickscript/tickscript.flux:268]:
```suggestion
// - column: Column to operate on. Default is _value.
// - fn: Aggregate or selector function to apply.
// - as: New column name.
// - every: Duration of windows.
// - defaultValue: Default fill value for null values in column.
//   Must be the same data type as column.
// - tables: Input data. Default is piped-forward data (`<-`).
```

File: stdlib/contrib/bonitoo-io/tickscript/tickscript.flux [hunk 3]
```diff
@@ -118,12 +247,41 @@ select = (column="_value", fn=(column, tables=<-) => tables, as, tables=<-) => {
...
+// ```
+// import "contrib/bonitoo-io/tickscript"
+//
+// data
+//   |> tickscript.selectWindow(
+//     fn: sum,
+//     as: "example-name",
+//     every: 1h,
+//     defaultValue: 0.0
+//   )
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/tickscript/tickscript.flux:281]:
```suggestion
// import "contrib/bonitoo-io/tickscript"
// # import "sampledata"
// # 
// # data = sampledata.int()
// #    |> range(start: sampledata.start, stop: sampledata.stop)
// 
// < data
//     |> tickscript.selectWindow(
//         fn: sum,
//         as: "example-name",
//         every: 1h,
//         defaultValue: 0,
// >     )
```

File: stdlib/contrib/bonitoo-io/tickscript/tickscript.flux [hunk 4]
```diff
@@ -141,23 +299,72 @@ selectWindow = (
         |> rename(fn: (column) => if column == _column then _as else column)
 }
 
-// compute computes aggregated value of the input data.
-// It is a convenience function to be used as
+// compute is an alias for tickscript.select() that
+// changes a column’s name and optionally applies an aggregate or selector
+// function.
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/tickscript/tickscript.flux:304]:
```suggestion
// compute is an alias for `tickscript.select()` that changes a column’s name and
// optionally applies an aggregate or selector function.
```

File: stdlib/contrib/bonitoo-io/tickscript/tickscript.flux [hunk 5]
```diff
@@ -141,23 +299,72 @@ selectWindow = (
...
 //
-//   |median('x)'
-//      .as(y)
+// ## Parameters
 //
+// - as: (Required) New column name.
+// - column: Column to operate on. Default is `_value`.
+// - fn: [Aggregate](https://docs.influxdata.com/flux/v0.x/function-types/#aggregates) or
+//   [selector](https://docs.influxdata.com/flux/v0.x/function-types/#selectors)
+//   function to apply.
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/tickscript/tickscript.flux:312]:
```suggestion
// - as: New column name.
// - column: Column to operate on. Default is `_value`.
// - fn: Aggregate or selector function to apply.
// - tables: Input data. Default is piped-forward data (`<-`).
//
// tags: transformations
```

File: stdlib/contrib/bonitoo-io/tickscript/tickscript.flux [hunk 6]
```diff
@@ -141,23 +299,72 @@ selectWindow = (
...
 compute = select
 
-// groupBy groups by specified columns.
-// It is a convenience function, it adds _measurement column which is required by monitor.check().
+// groupBy groups results by the `_measurement` column and other specified columns.
+//
+// This function is comparable to [Kapacitor QueryNode .groupBy](https://docs.influxdata.com/kapacitor/v1.6/nodes/query_node/#groupby).
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/tickscript/tickscript.flux:317]:
```suggestion
// This function is comparable to [Kapacitor QueryNode .groupBy](https://docs.influxdata.com/kapacitor/latest/nodes/query_node/#groupby).
```

File: stdlib/contrib/bonitoo-io/tickscript/tickscript.flux [hunk 7]
```diff
@@ -141,23 +299,72 @@ selectWindow = (
...
+//
+// (To group by intervals of time, use `window()` or `tickscript.selectWindow()`.)
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/tickscript/tickscript.flux:319]:
```suggestion
// **Note**: To group by time intervals, use `window()` or `tickscript.selectWindow()`.
```

File: stdlib/contrib/bonitoo-io/tickscript/tickscript.flux [hunk 8]
```diff
@@ -141,23 +299,72 @@ selectWindow = (
...
+//
+// ## Parameters
+// - columns: (Required) List of columns to group by.
+//
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/tickscript/tickscript.flux:323]:
```suggestion
// ## Parameters
// - columns: List of columns to group by.
// - tables: Input data. Default is piped-forward data (`<-`).
//
```

File: stdlib/contrib/bonitoo-io/tickscript/tickscript.flux [hunk 9]
```diff
@@ -141,23 +299,72 @@ selectWindow = (
...
+// ## Examples
+// ### Group by host and region
+// import "contrib/bonitoo-io/tickscript"
+// ```
+// data
+//   |> tickscript.groupBy(
+//     columns: ["host", "region"]
+//   )
+// ```
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/tickscript/tickscript.flux:332]:
```suggestion
// ### Group by host and region
// ```
// # import "array"
// import "contrib/bonitoo-io/tickscript"
// 
// # data = array.from(
// #     rows: [
// #         {_time: 2021-01-01T00:00:00Z, _measurement: "m", host: "h1", region: "east", _field: "foo", _value: 1.2},
// #         {_time: 2021-01-01T00:01:00Z, _measurement: "m", host: "h1", region: "east", _field: "foo", _value: 3.4},
// #         {_time: 2021-01-01T00:00:00Z, _measurement: "m", host: "h2", region: "east", _field: "foo", _value: 2.3},
// #         {_time: 2021-01-01T00:01:00Z, _measurement: "m", host: "h2", region: "east", _field: "foo", _value: 5.6},
// #         {_time: 2021-01-01T00:00:00Z, _measurement: "m", host: "h3", region: "west", _field: "foo", _value: 1.2},
// #         {_time: 2021-01-01T00:01:00Z, _measurement: "m", host: "h3", region: "west", _field: "foo", _value: 3.4},
// #         {_time: 2021-01-01T00:00:00Z, _measurement: "m", host: "h4", region: "west", _field: "foo", _value: 2.3},
// #         {_time: 2021-01-01T00:01:00Z, _measurement: "m", host: "h4", region: "west", _field: "foo", _value: 5.6},
// #     ],
// # )
// # 
// < data
// >     |> tickscript.groupBy(columns: ["host", "region"])
// ```
//
// tags: transformations
```

File: stdlib/contrib/bonitoo-io/tickscript/tickscript.flux [hunk 10]
```diff
@@ -141,23 +299,72 @@ selectWindow = (
...
     |> group(columns: columns)
     // required by monitor.check
     |> experimental.group(columns: ["_measurement"], mode: "extend")
 
-// join merges two streams using standard join().
-// It is meant a convenience function, it ensures _measurement column exists and is in the group key.
+// join merges two input streams into a single output stream
+// based on specified columns with equal values and appends a new measurement name.
+//
+// This function is comparable to [Kapacitor JoinNode](https://docs.influxdata.com/kapacitor/v1.6/nodes/join_node/).
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/tickscript/tickscript.flux:341]:
```suggestion
// This function is comparable to [Kapacitor JoinNode](https://docs.influxdata.com/kapacitor/latest/nodes/join_node/).
```

File: stdlib/contrib/bonitoo-io/tickscript/tickscript.flux [hunk 11]
```diff
@@ -141,23 +299,72 @@ selectWindow = (
...
+//
+// ## Parameters
+//
+// - tables: - record -(Required) Map of two streams to join.
+// - on: - array of strings - List of columns to join on. Default is `["_time"]`.
+// - measurement: - string -(Required) Measurement name to use in results.
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/tickscript/tickscript.flux:347]:
```suggestion
// - tables: Map of two streams to join.
// - on: List of columns to join on. Default is `["_time"]`.
// - measurement: Measurement name to use in results.
```

File: stdlib/contrib/bonitoo-io/tickscript/tickscript.flux [hunk 12]
```diff
@@ -141,23 +299,72 @@ selectWindow = (
...
+// states = //...
+//
+// tickscript.join(
+//   tables: {metric: metrics, state: states},
+//   on: ["_time", "host"],
+//   measurement: "example-m"
+// )
+//```
+//
+// [OUTPUT DATA]
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/tickscript/tickscript.flux:367]:
```suggestion
// ```
// import "array"
// import "contrib/bonitoo-io/tickscript"
// 
// metrics = array.from(
//     rows: [
//         {_time: 2021-01-01T00:00:00Z, host: "host1", _value: 1.2},
//         {_time: 2021-01-01T01:00:00Z, host: "host1", _value: 0.8},
//         {_time: 2021-01-01T02:00:00Z, host: "host1", _value: 3.2},
//         {_time: 2021-01-01T00:00:00Z, host: "host2", _value: 8.4},
//         {_time: 2021-01-01T01:00:00Z, host: "host2", _value: 7.3},
//         {_time: 2021-01-01T02:00:00Z, host: "host2", _value: 7.9},
//     ],
// )
//     |> group(columns: ["host"])
// 
// states = array.from(
//     rows: [
//         {_time: 2021-01-01T00:00:00Z, host: "host1", _value: "dead"},
//         {_time: 2021-01-01T01:00:00Z, host: "host1", _value: "dead"},
//         {_time: 2021-01-01T02:00:00Z, host: "host1", _value: "alive"},
//         {_time: 2021-01-01T00:00:00Z, host: "host2", _value: "alive"},
//         {_time: 2021-01-01T01:00:00Z, host: "host2", _value: "alive"},
//         {_time: 2021-01-01T02:00:00Z, host: "host2", _value: "alive"},
//     ],
// )
//     |> group(columns: ["host"])
// 
// tickscript.join(
//     tables: {metric: metrics, state: states},
//     on: ["_time", "host"],
//     measurement: "example-m",
// > )
//```
//
// tags: transformations
```

**Review by sanderson — CHANGES_REQUESTED** [review]

**Review by scbrickley — APPROVED** [review]

File: stdlib/contrib/bonitoo-io/tickscript/tickscript.flux [hunk 13]
```diff
@@ -49,7 +133,49 @@ alert = (
...
+//   Default is `(r) => "${r._check_id}"`.
+// - message: Function that returns the InfluxDB check message using data from input rows.
+//   Default is `(r) => "Deadman Check: ${r._check_name} is: " + (if r.dead then "dead" else "alive")`.
+// - topic: Check topic. Default is `""`.
+// - tables: Input data. Default is piped-forward data (`<-`).
+//
+// ## Examples
+//
+// ### Detect when a series stops reporting
+// ```
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/tickscript/tickscript.flux:162]:
```suggestion
// ```no_run
```

File: stdlib/contrib/bonitoo-io/tickscript/tickscript.flux [hunk 14]
```diff
@@ -141,23 +355,107 @@ selectWindow = (
...
-// groupBy groups by specified columns.
-// It is a convenience function, it adds _measurement column which is required by monitor.check().
+// groupBy groups results by the `_measurement` column and other specified columns.
+//
+// This function is comparable to [Kapacitor QueryNode .groupBy](https://docs.influxdata.com/kapacitor/latest/nodes/query_node/#groupby).
+//
+// **Note**: To group by time intervals, use `window()` or `tickscript.selectWindow()`.
+//
+// ## Parameters
+// - columns: (Required) List of columns to group by.
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/tickscript/tickscript.flux:378]:
```suggestion
// - columns: List of columns to group by.
```

**Review by sanderson — APPROVED** [review]
Two suggestions and then it's good to go.

**Pull request merged**